brave-okhttp/src/main/java/com/github/kristofa/brave/okhttp/BraveTracingInterceptor.java

```diff
@@ -0,0 +1,238 @@
+  public static BraveTracingInterceptor create(Brave brave) {
+    return builder(brave).build();
+  }
+
+  /** Defaults to use {@link DefaultSpanNameProvider} */
+  public static Builder builder(Brave brave) {
+    return new Builder(brave);
+  }
+
+  public static final class Builder {
+    Brave brave;
+    String serverName = "";
+    SpanNameProvider spanNameProvider = new DefaultSpanNameProvider();
+    boolean tagIsLocalSpanName = false;
+
+    Builder(Brave brave) {
+      this.brave = checkNotNull(brave, "brave");
+    }
+
+    /**
+     * Indicates the service name used for the {@link Constants#SERVER_ADDR server address}.Default
+     * is empty string.
+     *
+     * <p>Setting this is not important when the server is instrumented with Zipkin. This is
+     * important when the server is not instrumented with Zipkin. For example, if you are calling a
+     * cloud service, you will see this name as a leaf in your service dependency graph.
+     */
+    public Builder serverName(String serverName) {
+      this.serverName = checkNotNull(serverName, "serverName");
+      return this;
+    }
+
+    /**
+     * Indicates the span name of http requests. When {@link #tagIsLocalSpanName} is enabled, the
+     * {@link Request#tag()} is preferred for the application request.
+     */
+    public Builder spanNameProvider(SpanNameProvider spanNameProvider) {
+      this.spanNameProvider = checkNotNull(spanNameProvider, "spanNameProvider");
+      return this;
+    }
+
+    /**
+     * When enabled and a {@link Request#tag()} exists, it will be used as the local span name.
+     *
+     * <p>The local span is the one representing the application request. It has a child for each
+     * network request. These span name of network requests are unaffected by this setting.
+     */
+    public Builder tagIsLocalSpanName(boolean tagIsLocalSpanName) {
+      this.tagIsLocalSpanName = tagIsLocalSpanName;
+      return this;
+    }
+
+    public BraveTracingInterceptor build() {
+      return new BraveTracingInterceptor(this);
+    }
+  }
+
+  final Brave brave;
+  final ClientRequestInterceptor requestInterceptor;
+  final ClientResponseInterceptor responseInterceptor;
+  final SpanNameProvider spanName;
+  final boolean tagIsLocalSpanName;
+  final String serverName;
+
+  BraveTracingInterceptor(Builder builder) {
+    brave = builder.brave;
+    requestInterceptor = new ClientRequestInterceptor(brave.clientTracer());
+    responseInterceptor = new ClientResponseInterceptor(brave.clientTracer());
+    spanName = builder.spanNameProvider;
+    tagIsLocalSpanName = builder.tagIsLocalSpanName;
+    serverName = builder.serverName;
+  }
+
+  @Override
+  public Response intercept(Chain chain) throws IOException {
+    Request request = chain.request();
+    Connection connection = chain.connection();
+    if (connection == null) { // we are an application interceptor, so start a local span.
```

Comment:
Neat. I've never seen a single interceptor work in both modes, but it makes sense!